tests.py

```diff
@@ -1238,46 +1227,114 @@ def tearDown(self):
         self.assertTrue(self.r2 != self.r4)
-
         self.assertTrue(self.r3 != self.r4)
 
-    def test_equal_redis_dict(self):
-        self.assertEqual(self.r1, self.r2)
+    def test_neq_with_redis_dict(self):
         self.assertNotEqual(self.r1, self.r3)
         self.assertNotEqual(self.r1, self.r4)
 
-    def test_equal_with_dict(self):
-        self.assertEqual(self.r1, self.d1)
+    def test_neq_with_dict(self):
         self.assertNotEqual(self.r1, self.d2)
 
-    def test_empty_equal(self):
-        empty_r = RedisDict(namespace="test_empty")  # TODO make sure it's deleted
-        self.assertEqual(empty_r, {})
+    def test_neq_empty(self):
+        empty_r = RedisDict(namespace="test_empty")
+        self.assertNotEqual(self.r1, {})
+        self.assertNotEqual(empty_r, self.d1)
+        empty_r.clear()
 
-    def test_nested_empty_equal(self):
-        nested_empty_r = RedisDict(namespace="test_nested_empty")  # TODO make sure it's deleted
+    def test_neq_nested_empty(self):
+        nested_empty_r = RedisDict(namespace="test_nested_empty")
         nested_empty_r.update({"a": {}})
         nested_empty_d = {"a": {}}
-        self.assertEqual(nested_empty_r, nested_empty_d)
+        self.assertNotEqual(self.r1, nested_empty_d)
+        self.assertNotEqual(nested_empty_r, self.d1)
+        nested_empty_r.clear()
+
+    def test_is_comparison(self):
+        self.assertTrue(self.r1 is self.r1)
+        self.assertFalse(self.r1 is self.r2)
+        self.assertTrue(self.r2 is self.r2)
+        self.assertFalse(self.r2 is self.r3)
+
+    def test_is_comparison_with_dict(self):
+        self.assertFalse(self.r1 is self.d1)
+        self.assertFalse(self.d2 is self.d1)
+
+    def test_is_not_comparison(self):
+        self.assertFalse(self.r1 is not self.r1)
+        self.assertTrue(self.r1 is not self.r2)
+        self.assertFalse(self.r2 is not self.r2)
+        self.assertTrue(self.r2 is not self.r3)
+
+    def test_is_not_comparison_with_dict(self):
+        self.assertTrue(self.r1 is not self.d1)
+        self.assertTrue(self.d2 is not self.d1)
+
+    def test_lt(self):
+        with self.assertRaises(TypeError):
+            self.r1 < self.r2
+
+    def test_lt_with_different_type(self):
+        with self.assertRaises(TypeError):
+            self.r1 < self.d1
+
+    def test_le(self):
+        with self.assertRaises(TypeError):
+            self.r1 <= self.r2
+
+    def test_le_with_different_type(self):
+        with self.assertRaises(TypeError):
+            self.r1 <= self.d1
+
+    def test_ge(self):
+        with self.assertRaises(TypeError):
+            self.r1 >= self.r2
+
+    def test_ge_with_different_type(self):
+        with self.assertRaises(TypeError):
+            self.r1 >= self.d1
+
+    def test_gt(self):
+        with self.assertRaises(TypeError):
+            self.r1 > self.r2
+
+    def test_gt_with_different_type(self):
+        with self.assertRaises(TypeError):
+            self.r1 > self.d1
```

Comment:
line 1337 :)

Great work with the PR and the unit tests. However, I would like to implore you to add more tests. For instance, as example, your comment outlined in the [issue](https://github.com/Attumm/redis-dict/issues/38#issuecomment-1936065684) came really close.

So, although we already have tests in place, what if we added tests that are very simple to read, using two dictionaries and one Redis dictionary.

If we systematically perform each comparison operator on them and modify them in sequential steps, we could ensure uniform behaviour and output across all operators.

Please consider the following, using assert for the sake of clarity.
```python
# Empty dictionaries are created
d = {}
d2 = {} # d2 is used as the expected.
rd = redisdict()

# Testing for equality
assert d == d2
assert d == rd


d["foo"] = "bar"

# Testing for inequality after modification in 'd'
assert d != d2
assert d != rd

# Modifying 'd2' and 'rd'
d2["foo"] = "bar"
rd["foo"] = "bar"

# Testing for equality after modifications in 'd2' and 'rd'
assert d == d2
assert d == rd
```

Reply:
okay i wasnt sure how you would prefer it as there were already some existing tests which i tried to reuse, but this sounds good! I will improve them a bit more for clarity and readability.

Reply:
Ive just added a sequential test that follows the ideas and structure as described. I was not sure if you would prefer it readable as one test or split up into smaller tests so i hope this is alright :)